pipelines/controller/src/main/java/org/openmrs/analytics/ApiController.java

```diff
@@ -78,4 +90,21 @@ private Stats getStats() {
     MetricQueryResults metricQueryResults = pipelineManager.getMetricQueryResults();
     return Stats.createStats(metricQueryResults);
   }
+
+  @GetMapping(
+      value = "/download",
+      produces = {MediaType.APPLICATION_OCTET_STREAM_VALUE})
+  public ResponseEntity<InputStreamResource> download(@RequestParam(name = "path") String path)
+      throws IOException {
+    ResourceId resourceId = FileSystems.matchNewResource(path, false);
+    ReadableByteChannel channel = FileSystems.open(resourceId);
+    InputStream stream = Channels.newInputStream(channel);
+    InputStreamResource inputStreamResource = new InputStreamResource(stream);
+    MultiValueMap<String, String> headers = new HttpHeaders();
+    headers.put(
+        "Content-Disposition",
+        Arrays.asList("inline; filename=\"" + resourceId.getFilename() + "\""));
+    headers.put("Content-type", Arrays.asList(MediaType.APPLICATION_OCTET_STREAM_VALUE));
```

Comment:
I had kept `octet-stream` so that the API can be general which can be used to download files with any format. I have changed the type to `text/plain` now as currently we use it for text files only.